src/btree/bt_cursor.c

```diff
@@ -443,6 +443,7 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
 	} else if ((ret = __wt_btcur_next(cbt, false)) != WT_NOTFOUND)
 		exact = 1;
 	else {
+		WT_ERR(__cursor_func_init(cbt, true));
```

Comment:
The cursor is not positioned at this point, so we may have released our snapshot.  We need a new one to restart the search.